### PR TITLE
#7266 Remove configuration holder metrics

### LINKED ISSUE #7266: Remove `ConfigurationHolder` from `swirlds-common`
Remove `ConfigurationHolder` from the `swirlds-common` module.

### PR DESCRIPTION
**Pull request by timo0**
Fix #7266

### CODE REVIEW

File: platform-sdk/swirlds-benchmarks/build.gradle.kts [hunk 1]
```diff
@@ -15,30 +15,33 @@
...
+    id("com.swirlds.platform.library")
+    id("com.swirlds.platform.benchmark-conventions")
 }
 
 dependencies {
-  // JMH Dependencies
-  jmhImplementation(project(":swirlds-platform-core"))
-  jmhImplementation(project(":swirlds-config-api"))
-  jmhImplementation(project(":swirlds-config-impl"))
+    implementation(project(mapOf("path" to ":swirlds-unit-tests:common:swirlds-test-framework")))
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-benchmarks/build.gradle.kts:24]:
```suggestion
    implementation(project(:swirlds-unit-tests:common:swirlds-test-framework"))
```

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/metrics/RunningAverageMetric.java [hunk 2]
```diff
@@ -106,16 +105,17 @@ final class Config extends MetricConfig<RunningAverageMetric, RunningAverageMetr
...
+         *      the metrics configuration
          * @param category
          * 		the kind of metric (stats are grouped or filtered by this)
          * @param name
          * 		a short name for the statistic
          * @throws IllegalArgumentException
          * 		if one of the parameters is {@code null} or consists only of whitespaces
          */
-        public Config(final String category, final String name) {
+        public Config(final MetricsConfig metricsConfig, final String category, final String name) {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/metrics/RunningAverageMetric.java:117]: null annotations missing for all the inner Config constructors

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/FCMQueryController.java [hunk 3]
```diff
@@ -48,18 +51,19 @@ class FCMQueryController {
...
-            .withFormat(FORMAT_9_6);
     private final SpeedometerMetric fcmQueriesAnsweredPerSecond;
 
     public FCMQueryController(final FCMConfig.FCMQueryConfig fcmQueryConfig, final Platform platform) {
         this.platform = platform;
         this.numberOfThreads = fcmQueryConfig.getNumberOfThreads();
         this.throttle = new Throttle(fcmQueryConfig.getQps());
-        this.fcmQueriesAnsweredPerSecond =
-                platform.getContext().getMetrics().getOrCreate(FCM_QUERIES_ANSWERED_PER_SECOND_CONFIG);
+        final Configuration configuration = platform.getContext().getConfiguration();
```
Comment (hendrikebbers) [inline platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/FCMQueryController.java:60]: With the given pattern that you introduce the `MetricsConfig` is passed to the individual creation of each `Metric`. I would have expected that the `MetricsConfig` is initially passed only once to the `Metrics` class.

**Review by hendrikebbers — COMMENTED** [review]
I do not know if that is the best pattern. Why not passing the config to the creation of the Metrics

**Comment by timo0** [comment]
Closed because of new approach

**Pull request closed without merging**